[coqdep] moved coqdeplib into own directory

Alizter opened the pull request:
We do some refactoring on coqdep. Notably, we separate the parsing and initialisation phases of coqdep, clearing the ambiguity with coqdeplib that was there before.

This will allow for further cleanup and refactoring.

cc @ejgallego

ejgallego reviewed (commented):
Looks good, maybe the directory could just be `coqdep/lib` ?

Having a top-level error and args module is fine, however not very useful to define interfaces for them if in the end they are called from only one module.

In general, I think it makes sense to split a module to its own file when it has more than one user, large surface API has a large cost IMHO.

Alizter commented:
@ejgallego I've renamed coqdeplib to lib and removed the toplevel error interface. However the args module is important since this is what you pass to initialize coqdep.

We can do some more cleanups of what is in the API soon when we begin using it ourselves.

ejgallego commented on tools/coqdep/lib/args.ml:
```diff
@@ -0,0 +1,93 @@
+(************************************************************************)
+
+type t =
+{ boot : bool
```
Is the indentation here ok?

Alizter replied:
Missed that thanks.

ejgallego commented on tools/coqdep/lib/options.mli:
```diff
@@ -8,4 +8,12 @@
+val sort : bool ref
+val write_vos : bool ref
+val noglob : bool ref
+val dynlink : Dynlink.t ref
```
At some point in a future PR we could remove these and have the clients use `Args.t` directly.

Alizter replied:
@ejgallego They can initialize Coqdep.Lib with Args.t directly, thats what init does now. These options are simply the internal state of coqdep. We could probably move this to a common state file.

ejgallego commented:
Will merge soon, unless you want to squash some history.

Alizter commented:
I won't squash this since it is tricky to undo these commits otherwise.

ejgallego commented:
Yeah no need to squash, however for example the renaming and the indentation fix could disappear easily using `rebase -i` and make a bit better history.

Alizter commented:
@ejgallego ping

ejgallego reviewed (approved):
Thanks

ejgallego commented:
@coqbot: merge now
